prometheus-metric-storage/src/lib.rs

```diff
@@ -107,6 +118,8 @@ impl StorageRegistry {
                 }
             };
 
+        // nlordell: nit: Maybe put the `unsafe` only around the unsafe part,
+        // so here turning a `'_` reference into a `'static` one.
```

Comment:
So I'm not sure what are the rules around unsafe blocks. When I was reading the nomicon, I got the impression that unsafe blocks span all code that can potentially break the invariant. On the other hand, I like the idea of small unsafe blocks that only highlight actual unsafe calls. So I think I'll implement this nit 😊

Also note that we're not turning it into a `'static` reference, but rather into a reference with the lifetime of `&self`. This is because the signature of this function is equivalent to:

```rust
pub fn get_storage<'a, T>(&'a self, ...) -> Result<&'a T>
```

I'll add a comment explaining this.

Reply:
I think it is good style to have unsafe blocks as small as possible. This is to make it clear which exact operation is unsafe and would usually not be allowed by the compiler. Prevents accidentally adding other unsafe parts where you didn't want to.

Reply:
> so here turning a `'_` reference into a `'static` one.

Yeah, I was a bit confused about things :see_no_evil: